File: 3105. Longest Strictly Increasing or Strictly Decreasing Subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        ans, count, n = 1, 1, len(nums)
        for i in range(1, n):
            if nums[i] <= nums[i - 1]:
                count = 0
            count += 1
            ans = max(ans, count)

        count = 1
        for i in range(1, n):
            if nums[i] >= nums[i - 1]:
                count = 0
            count += 1
            ans = max(ans, count)

        return ans

File: 3105. Longest Strictly Increasing or Strictly Decreasing Subarray.cpp
class Solution {
public:
    int ok(vector<int>&a)
    {
        int n = a.size();
        int mx = 1, cur = 1;
        for(int i=1;i<n;i++)
        {
            if(a[i]<a[i-1]) cur++;
            else 
            {
                mx = max(mx, cur);
                cur = 1;
            }
        }
        return max(cur, mx);
    }
    int ok2(vector<int>&a)
    {
        int n = a.size();
        int mx = 1, cur = 1;
        for(int i=1;i<n;i++)
        {
            if(a[i]>a[i-1]) cur++;
            else 
            {
                mx = max(mx, cur);
                cur = 1;
            }
        }
        return max(cur, mx);
    }
    int longestMonotonicSubarray(vector<int>& nums) 
    {
        return max(ok(nums), ok2(nums));
    }
};